src/M6Web/Bundle/LogBridgeBundle/Matcher/Status/Type/AbstractType.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace M6Web\Bundle\LogBridgeBundle\Matcher\Status\Type;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * Abstract class AbstractType
+ */
+abstract class AbstractType implements TypeInterface
+{
+    /**
+     * @var string
+     */
+    protected $regex = null;
+
+    /**
+     * Type Constructor
+     */
+    public function __construct()
+    {
+        if ($this->regex === null) {
```

Comment:
I would prefer an abstract method getRegex (or Maybe getPattern ?) to be defined in the concrete class - so no if, no exception etc.